File: interface/matlab/thalassa_interface_example.m
% Add path for MEX library
addpath("../../lib/")

% Set initial state
state.mjd = 59043.0;
state.RV = [7000, 0, 0, 0, 8, 0];

% Set model parameters
model.insgrav = 1;
model.isun    = 1;
model.imoon   = 1;
model.idrag   = 1;
model.iF107   = 1;
model.iSRP    = 2;
model.iephem  = 2;
model.gdeg    = 2;
model.gord    = 2;

% Set filepaths
paths.phys_path   = '../../data/physical_constants.txt';
paths.earth_path  = '../../data/earth_potential/GRIM5-S1.txt';
paths.kernel_path = '../../data/kernels_to_load.furnsh';

% Set propagator settings
settings.tol    = 1e-8;
settings.tspan  = 365.25;
settings.tstep  = 1.0;
settings.mxstep = 1e6;
settings.imcoll = 1;
settings.eqs    = 2;

% Set spacecraft
spacecraft.mass      = 1500;
spacecraft.area_drag = 15;
spacecraft.area_srp  = 15;
spacecraft.cd        = 2.2;
spacecraft.cr        = 1.5;

% Execute THALASSA via the MEX interface
[times, states] = mthalassa(state, model, paths, settings, spacecraft);

File: interface/cpp/thalassa_interface_example.cpp
#include <cthalassa/cthalassa.hpp>

int main() {
    cthalassa::PropagatorModel model;

    cthalassa::PropagatorPaths paths = {"./data/physical_constants.txt", "./data/earth_potential/GRIM5-S1.txt", "./data/kernels_to_load.furnsh"};

    cthalassa::PropagatorSettings settings;

    cthalassa::PropagatorSpacecraft spacecraft = {+8500.000000000000E+00, +13.00000000000000E+00, +13.00000000000000E+00, +2.200000000000000E+00,
                                                  +1.500000000000000E+00};

    cthalassa::Propagator propagator(model, paths, settings, spacecraft);

    std::vector<double> stateIn = {7.000000000000000E+03, 0.000000000000000E+00, 0.000000000000000E+00,
                                   0.000000000000000E+00, 7.546053286792848E+00, 0.000000000000000E+00};

    double tStart = 59043;
    double tEnd = tStart + 7;
    double tStep = 1;

    std::vector<double> timesOut;
    std::vector<std::vector<double>> statesOut;

    propagator.propagate(tStart, tEnd, tStep, stateIn, timesOut, statesOut);

    return 0;
}